index.js

```diff
@@ -218,11 +266,11 @@
   //. > S.Just(42).empty()
   //. Nothing()
   //. ```
-  Maybe.prototype.empty = Maybe.empty;
+  Maybe.prototype.empty = def('Maybe#empty', [], Maybe.empty);
 
-  //# Maybe#equals :: Maybe a ~> Maybe a -> Boolean
+  //# Maybe#equals :: Maybe a ~> b -> Boolean
   //.
-  //. Takes a Maybe and returns `true` if:
+  //. Takes a value of any type and returns `true` if:
```

Comment:
I realized that the current type is too restrictive. `S.Just(42).equals([1, 2, 3])` should evaluate false rather than throw. The current implementation does not type check its argument; this simply updates the documentation to match the actual behaviour. I also added tests to ensure that the argument needn't be a Maybe.